src/app/components/find-journey/find-journey.component.ts

```diff
@@ -8,15 +14,79 @@ import {Component, OnInit} from '@angular/core';
   templateUrl: './find-journey.component.html',
   styleUrls: ['./find-journey.component.scss']
 })
-export class FindJourneyComponent implements OnInit {
+export class FindJourneyComponent {
 
-  constructor() { }
+  journey: Journey;
+  journeysFound: Journey[];
+  destination = new FormControl('');
+  id = new FormControl('');
 
-  ngOnInit() {
+  destination$: Observable<any[]>;
+  loading$: Observable<boolean>;
+  errors$: Observable<any>;
+
+  constructor(private apollo: Apollo) {
+    this.journey = new Journey();
+    this.journeysFound = [];
   }
 
   onSubmit() {
-    console.log('test');
+    console.log('id searched: ', this.id.value);
+    console.log('destination searched: ', this.destination.value);
+
+    this.searchById();
   }
 
+  searchById() {
+    const source$ = this.apollo.query({
+      query: gql`
+        query find{
+          findJourneyById(id: "tt") {
+            id,
+            destination
+          }
+        }
+      `
+    }).pipe(shareReplay(1));
+
+    this.destination$ = source$.pipe(map(result => result.data && result.data['findJourneyById']));
```

Comment:
I don't understand this line

Reply:
we do a sort of 'pluck' here.
cause we don't need all the info returned by graphql